src/components/ProfilePage.tsx

```diff
@@ -0,0 +1,217 @@
+    const [ location, setLocation ] = useState<string|undefined>()
+    const [ summary, setSummary ] = useState<string|undefined>()
+    const [ url, setUrl ] = useState<string|undefined>()
+    const [ profilePicUrl, setProfilePicUrl ] = useState<string|undefined>()
+    const [ fileHandleId, setFileHandleId ] = useState<string|undefined>()
+    const [ verified, setVerfied ] = useState<boolean>()
+    const [ editing, setEditing ] = useState(false)
+
+    const getProfile = async () => {
+        try{
+            const mask =
+            SynapseConstants.USER_BUNDLE_MASK_IS_VERIFIED |
+            SynapseConstants.USER_BUNDLE_MASK_USER_PROFILE
+
+            const bundle: UserBundle = await getMyUserBundle(mask, accessToken)
+            const picUrl = await getFileHandleByIdURL(bundle.userProfile?.profilePicureFileHandleId as string,accessToken)
+            setUserProfile(bundle.userProfile)
+            setFirstName(bundle.userProfile?.firstName)
+            setLastName(bundle.userProfile?.lastName)
+            setPosition(bundle.userProfile?.position)
+            setCompany(bundle.userProfile?.company)
+            setLocation(bundle.userProfile?.location)
+            setSummary(bundle.userProfile?.summary)
+            setUrl(bundle.userProfile?.url)
+            setProfilePicUrl(picUrl)
+            setVerfied(bundle.isVerified)
+        } catch(err:any) {
+            displayToast(err.reason as string, 'danger')
+        }
+    }
+
+    const updateUserProfile = async () => {
+        try{
+            if(userProfile){
+                userProfile.firstName = firstName as string
+                userProfile.lastName = lastName as string
+                userProfile.position = position as string
+                userProfile.company = company as string
+                userProfile.location = location as string
+                userProfile.summary = summary as string
+                userProfile.url = url as string
+                userProfile.profilePicureFileHandleId = fileHandleId as string
+                await updateMyUserProfile(userProfile, accessToken)
+                displayToast('Profile has been successfully updated', 'success')
+                getProfile()
+                setEditing(false)
+            }
+        } catch(err:any){
+            displayToast(err.reason as string, 'danger')
+        }
+    }
+
+    useEffect(()=>{
+        getProfile()
+    },[accessToken,])
+
+    const EditField = (label: string, updatedValue: string|undefined, updateFn: Function) => {
+        return(
+            <div className='edit-field'>
+                <FormLabel>{label}</FormLabel>
+                <FormControl
+                    onChange={e=>updateFn(e.target.value)}
+                    value={updatedValue}
+                    placeholder= {`Enter ${label}`}
+                /> 
+            </div>
+        )
+    }
+
+    const onCancel = async () => {
+        getProfile()
```

Comment:
Well, this could be refactored so that you save the user bundle in the state, and in this case revert values to what we found in the bundle.  The same code should be used in both places (for initial population of other state variables, and for this reset).